lib/iris/cube.py

```diff
@@ -758,6 +767,231 @@ def _is_single_item(testee):
+
+    @locals.setter
+    def locals(self, attributes):
+        self._locals = self._normalise_attrs(attributes)
+
+    @property
+    def globals(self):
+        return self._globals
+
+    @globals.setter
+    def globals(self, attributes):
+        self._globals = self._normalise_attrs(attributes)
+
+    #
+    # Provide a serialisation interface
+    #
+    def __getstate__(self):
+        return (self.locals, self.globals)
+
+    def __setstate__(self, state):
+        self.locals, self.globals = state
+
+    #
+    # Support simple comparison, even when contents are arrays.
+    #
+    def __eq__(self, other):
+        # For equality, require both globals + locals to match
+        other = CubeAttrsDict(other)
+        result = self.locals == other.locals and self.globals == other.globals
+        return result
+
+    def __ne__(self, other):
+        return not self == other
+
+    #
+    # Provide a copy method, as for 'dict', but *not* provided by MutableMapping
+    #
+    def copy(self):
+        """
+        Return a copy.
+
+        Implemented with deep copying, consistent with general Iris usage.
+
+        """
+        return CubeAttrsDict(self)
+
+    #
+    # The remaining methods are sufficient to generate a complete standard Mapping
+    # API -- see docs for :class:`collections.abc.MutableMapping`.
+    #
+
+    def __iter__(self):
+        # Ordering: all global keys, then all local ones, but omitting duplicates.
+        # NOTE: this means that in the "summary" view, attributes present in both
+        # locals+globals are listed first, amongst the globals, even though they appear
+        # with the *value* from locals.
+        # Otherwise follows order of insertion, as is normal for dicts.
+        return itertools.chain(
+            self.globals.keys(),
+            (x for x in self.locals.keys() if x not in self.globals),
+        )
+
+    def __len__(self):
+        # Return the number of keys in the 'combined' view.
+        return len(list(iter(self)))
+
+    def __getitem__(self, key):
+        # Fetch an item from the "combined attributes".
+        # If both present, the local setting takes priority.
+        if key in self.locals:
+            store = self.locals
+        else:
+            store = self.globals
+        return store[key]
+
+    def __setitem__(self, key, value):
+        # Assign an attribute value.
+        #  Make local or global according to the existing content, or the known set of
+        # "normally global" attributes given by CF.
+
+        # If an attribute of this name is already present, update that
+        # (the local one having priority).
+        if key in self.locals:
+            store = self.locals
+        elif key in self.globals:
+            store = self.globals
+        else:
+            # If NO existing attribute,  create local unless it is a "known global" one.
+            from iris.fileformats.netcdf.saver import _CF_GLOBAL_ATTRS
+
+            # Not in existing
```

Comment:
This comment is unhelpful (!)
(probably left over from an earlier version)

Reply:
Fixed https://github.com/SciTools/iris/commit/9bbaf5559db77e271fad6c817e3129877598e8fe